helpers/misc.go

```diff
@@ -146,3 +147,28 @@ func MergeNonZero[T any](original T, overrides T) T {
 	}
 	return originalValue.Elem().Interface().(T)
 }
+
+// MergePathAndValueIntoMap takes a path in dot notation as a string and a value (also as a string for simplicity),
+// then merges this into the provided map. The value can be any type.
+func MergePathAndValueIntoMap(m map[string]any, path string, value any) error {
+	pathParts := strings.Split(path, ".")
+	current := m
+	for i, part := range pathParts {
+		if i == len(pathParts)-1 {
+			// Set the value at the last key in the path.
+			current[part] = value
+		} else {
+			if _, exists := current[part]; !exists {
+				// If the part does not exist, create a new map for it.
+				current[part] = make(map[string]any)
+			}
+
+			nextMap, ok := current[part].(map[string]any)
+			if !ok {
+				return fmt.Errorf("conflict at '%s', expected map but got %T", strings.Join(pathParts[:i+1], "."), current[part])
```

Comment:
```suggestion
				return fmt.Errorf("conflict at %q, expected map[string]any but got %T", strings.Join(pathParts[:i+1], "."), current[part])
```

`%q` will wrap strings in `"` for you, anything that implements Stringers and can be a `%s` can also be a `%q`.

Usage varies depending upon which one you want to use in a string.

Reply:
I took care of it.